### PR TITLE
Documentation for broadcasting behavior of binary elementwise operations

### PR DESCRIPTION
**Pull request by siddharth-agrawal**
PR for #508. Uses @mrry's example in the SO question referenced in the issue.

### CODE REVIEW

File: tensorflow/python/ops/math_ops.py [hunk 1]
```diff
@@ -13,7 +13,26 @@
...
 
-"""## Arithmetic Operators
+"""Note: Elementwise binary operations in TensorFlow follow [numpy-style
+broadcasting](http://docs.scipy.org/doc/numpy/user/basics.broadcasting.html).
+Let us take an example of `tf.mul()` which falls in the above mentioned
+category, to clarify broadcasting and also give example usage of
+`tf.expand_dims()` when broadcasting is desired.
+
+Ex: Let us say that we have a batch of images in the tensor `image_batch` which
+is of size `b x h x w x c`, and a mask tensor `image_mask` of size `h x w` that
```
Comment (vrv) [inline tensorflow/python/ops/math_ops.py]: 'which has shape [b, h, w, c], and as mask tensor of size '[h, w]' ...

File: tensorflow/python/ops/math_ops.py [hunk 2]
```diff
@@ -13,7 +13,26 @@
...
+we need to apply to all the images in the batch across all the channels. We thus
```
Comment (vrv) [inline tensorflow/python/ops/math_ops.py]: Do you think it's necessary to have additional text here?  Why isn't the link to the numpy page enough?  Thoughts @martinwicke  ?

Comment (martinwicke) [inline tensorflow/python/ops/math_ops.py]: I'd be done with just the reference to numpy... No example here is complete, and may actually be confusing.

Comment (siddharth-agrawal) [inline tensorflow/python/ops/math_ops.py]: @vrv @martinwicke Updated the PR.

**Pull request merged**